rest-service/manager_rest/rest/resources_v1/executions.py

```diff
@@ -201,6 +201,58 @@ def post(self, **kwargs):
                 execution.id = execution_id
             sm.put(execution)
             messages = []
+            if (
+                force_status in ExecutionState.STATES
+                and force_status not in ExecutionState.WAITING_STATES
+                and force_status != ExecutionState.PENDING
+            ):
+                if deployment_id:
```

Comment:
maybe this whole case could be in a separate method, to avoid growing this `def post` even more?
Like
```python
def _on_execution_restored(self, sm, execution, deployment=None):
```

Reply:
Agreed-ish- I've factored it out specifically for linked executions. if we have more things to deal with later then maybe we tweak it further.